v1/skyflow/vaultapi/detokenize.go

```diff
@@ -93,107 +93,107 @@ func (detokenize *DetokenizeApi) sendRequest(ctx context.Context, records common
-				}
-				detokenizeParameters = append(detokenizeParameters, detokenizeParameter)
-			}
-			var body = make(map[string]interface{})
-			body["detokenizationParameters"] = detokenizeParameters
-			requestBody, err := json.Marshal(body)
-			if err == nil {
-				var request *http.Request
-				if ctx != nil {
-					request, _ = http.NewRequestWithContext(
-						ctx,
-						"POST",
-						requestUrl,
-						strings.NewReader(string(requestBody)),
-					)
-				} else {
-					request, _ = http.NewRequest(
-						"POST",
-						requestUrl,
-						strings.NewReader(string(requestBody)),
-					)
-				}
-				bearerToken := fmt.Sprintf("Bearer %s", detokenize.Token)
-				request.Header.Add("Authorization", bearerToken)
-				skyMetadata := common.CreateJsonMetadata()
-				request.Header.Add("sky-metadata", skyMetadata)
-				res, err := Client.Do(request)
-				var requestId = ""
-				if res != nil {
-					requestId = res.Header.Get("x-request-id")
-				}
-				if err != nil {
-					logger.Error(fmt.Sprintf(messages.DETOKENIZING_BULK_FAILED, detokenizeTag, requestId))
-					var error = make(map[string]interface{})
-					var errorObj = make(map[string]interface{})
-					errorObj["code"] = "500"
-					errorObj["description"] = common.AppendRequestId(fmt.Sprintf(messages.SERVER_ERROR, detokenizeTag, err), requestId)
-					error["error"] = errorObj
-					responseChannel <- error
-					return
+	if ( !detokenize.Options.ContinueOnError ) {
+			go func( responseChannel chan map[string]interface{}) {
+				var detokenizeParameters []map[string]interface{}
+				for i := 0; i < len(records.Records); i++ {
+					logger.Info(fmt.Sprintf(messages.DETOKENIZING_RECORDS, detokenizeTag, records.Records[i].Token))
+					singleRecord := records.Records[i]
+					var detokenizeParameter = make(map[string]interface{})
+					detokenizeParameter["token"] = singleRecord.Token
+					if singleRecord.Redaction == "" {
+						detokenizeParameter["redaction"] = common.PLAIN_TEXT
+					} else {
+						detokenizeParameter["redaction"] = singleRecord.Redaction
+					}
+					detokenizeParameters = append(detokenizeParameters, detokenizeParameter)
 				}
-				data, _ := ioutil.ReadAll(res.Body)
-				defer res.Body.Close()
-				var result map[string]interface{}
-				err = json.Unmarshal(data, &result)
-				if err != nil {
-					logger.Error(fmt.Sprintf(messages.DETOKENIZING_BULK_FAILED, detokenizeTag, requestId))
-					var error = make(map[string]interface{})
-					var errorObj = make(map[string]interface{})
-					errorObj["code"] = "500"
-					errorObj["description"] = fmt.Sprintf(messages.UNKNOWN_ERROR, detokenizeTag, common.AppendRequestId(string(data), requestId))
-					error["error"] = errorObj
-					responseChannel <- error
-				} else {
-					errorResult := result["error"]
-					if errorResult != nil {
+				var body = make(map[string]interface{})
+				body["detokenizationParameters"] = detokenizeParameters
+				requestBody, err := json.Marshal(body)
+				if err == nil {
```

Comment:
Instead of wrapping the case when err == nil, can we write a condition 
if err!=nil
{
 return
}

This will avoid wrapping a big block in if statement